doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/examples/ml-example.md

```diff
@@ -1,10 +1,253 @@
 (kuberay-ml-example)=
 
-# Example machine learning workloads
+# XGBoost-Ray on Kubernetes
 
-:::{warning}
-This page is under construction!
+:::{note}
+To learn the basics of Ray on Kubernetes, we recommend taking a look
+at the {ref}`introductory guide<kuberay-quickstart>` first.
 :::
 
-At least one end-to-end example of an actual machine learning workload,
-preferably with GPUs, possibly engaging the autoscaling functionality.
+In this guide, we show you how to run a sample Ray machine learning
+workload on Kubernetes infrastructure.
+
+We will run Ray's {ref}`XGBoost training benchmark<xgboost-benchmark>` with a 100 gigabyte training set.
+* Learn more about {ref}`XGBoost-Ray<xgboost-ray>`.
+
+## A note on autoscaling
+This guide will show you how to run an XGBoost workload with and without optional Ray Autoscaler support.
+When choosing whether or not to use autoscaling functionality, keep the following considerations in mind.
+
+### Autoscaling: Pros
+
+#### Cope with unknown resource requirements.
+If you don't know how much compute your Ray workload will require,
+autoscaling can adjust your Ray cluster to the right size.
+
+#### Save on costs.
+Idle compute is automatically scaled down, potentially leading to cost savings.
+
+### Autoscaling: Cons
+
+#### Less predictable when resource requirements are known.
+If you already know exactly how much compute your workload requires, it makes
+sense to provision a statically-sized Ray cluster.
+In this guide's example, we know that we need 1 Ray head and 9 Ray workers,
+so autoscaling is not strictly required.
```

Comment:
In fact, there's no good reason to use the Ray autoscaler with this workload, but

a. Doesn't hurt to give examples of how the autoscaler works
b. I am relieved that autoscaling functions as expected with this workload